src/pybamm/expression_tree/operations/serialise.py

```diff
@@ -435,6 +435,13 @@ def serialise_custom_model(model: pybamm.BaseModel) -> dict:
                 str(variable_name): convert_symbol_to_json(expression)
                 for variable_name, expression in getattr(model, "variables", {}).items()
             },
+            "parameter_values": (
+                Serialise._serialise_parameter_values(
+                    getattr(model, "_parameter_values", None)
+                )
+                if getattr(model, "_parameter_values", None) is not None
+                else None
+            ),
```

Comment:
I don't think adding parameter values as an attribute to the model is the right pattern, I could see this causing some bugs. It would be better to have a "parameterized model" object which is model + parameter values.